src/main/java/spireMapOverhaul/patches/interfacePatches/EncounterModifierPatches.java

```diff
@@ -0,0 +1,145 @@
+            }
+            return result;
+        }
+    }
+
+    @SpirePatch2(clz = AbstractDungeon.class, method = "getEliteMonsterForRoomCreation")
+    public static class EliteEncounterPatch {
+        @SpirePostfixPatch
+        public static MonsterGroup ChangeEncounter(AbstractDungeon __instance, MonsterGroup __result) {
+            MonsterGroup result = __result;
+            AbstractZone zone = Wiz.getCurZone();
+            if (zone instanceof EncounterModifyingZone) {
+                EncounterModifyingZone z = (EncounterModifyingZone)zone;
+                List<EncounterModifyingZone.ZoneEncounter> encounters = z.getEliteEncounters();
+                if (encounters != null) {
+                    encounters = encounters.stream().filter(e -> e.getActNum() == ActUtil.getRealActNum()).collect(Collectors.toList());
+                    if (LastZoneEliteEncounter.lastZoneEliteEncounter != null && LastZoneEliteEncounter.lastZoneEliteEncounter.startsWith(zone.id + ":")) {
+                        String encounterId = LastZoneEliteEncounter.lastZoneEliteEncounter.substring(zone.id.length() + 1);
+                        encounters = encounters.stream().filter(e -> !e.getID().equals(encounterId)).collect(Collectors.toList());
+                    }
+                    if (encounters.isEmpty()) {
+                        LastZoneEliteEncounter.lastZoneEliteEncounter = null;
+                    }
+                    else {
+                        EncounterModifyingZone.ZoneEncounter ze = encounters.get(AbstractDungeon.monsterRng.random(encounters.size() - 1));
+                        AbstractDungeon.lastCombatMetricKey = ze.getID();
+                        LastZoneEliteEncounter.lastZoneEliteEncounter = zone.id + ":" + ze.getID();
+                        result = ze.getMonsterSupplier().get();
+                    }
+                }
+
+                addAdditionalMonsters((EncounterModifyingZone)zone, result);
+                result = z.changeEncounter(result);
+            }
+            return result;
+        }
+    }
+
+    private static void addAdditionalMonsters(EncounterModifyingZone zone, MonsterGroup monsterGroup) {
+        List<AbstractMonster> additionalMonsters = zone.getAdditionalMonsters();
+        if (additionalMonsters != null) {
+            float currentX = monsterGroup.monsters.stream().map(m -> m.drawX).min(Comparator.comparingDouble(x -> (double)x)).orElse(0.0f);
+            for (AbstractMonster monster : additionalMonsters) {
+                // We give healthy additional space to account for fights such as Reptomancer, which spawn minions in
+                // the empty space near the main monster. Since there are modded enemies with this behavior too, this
+                // isn't perfect, but it's pretty much impossible to get perfect positioning for this given all the
+                // modded enemies out there and other ways that enemies can be added to fights.
```

Comment:
I've seen various attempts to do this (positioning new enemies dynamically). If you know of one that you think is more robust than this, we can switch over to it.